test/generators/devise/devise_jwt_install_generator_test.rb

```diff
@@ -0,0 +1,100 @@
+# frozen_string_literal: true
+
+require "test_helper"
+require "generators/boring/devise/jwt/install/install_generator"
+
+class DeviseInstallGeneratorTest < Rails::Generators::TestCase
+  tests Boring::Devise::Jwt::InstallGenerator
+  setup :build_app
+  teardown :teardown_app
+
+  include GeneratorHelper
+  include ActiveSupport::Testing::Isolation
+
+  def destination_root
+    app_path
+  end
+
+  def test_should_exit_if_devise_is_not_installed
+    assert_raises SystemExit do
+      quietly { generator.verify_presence_of_devise_gem }
+    end
+  end
+
+  def test_should_exit_if_devise_initializer_is_not_present
+    assert_raises SystemExit do
+      quietly { generator.verify_presence_of_devise_initializer }
+    end
+  end
+
+  def test_should_exit_if_devise_model_is_not_present
+    assert_raises SystemExit do
+      quietly { generator.verify_presence_of_devise_model }
+    end
+  end
+
+  def test_should_install_devise_jwt_successfully
+    Dir.chdir(app_path) do
+      setup_devise
+      quietly { run_generator }
+      assert_gem "devise-jwt"
+      assert_file "config/initializers/devise.rb" do |content|
+        assert_match(/config.jwt do |jwt|/, content)
+        assert_match(/jwt\.secret\s*=\s*ENV\['DEVISE_JWT_SECRET'\]/, content)
+        assert_match(/jwt\.dispatch_requests\s*=\s*\[\s*/, content)
+        assert_match(/jwt\.revocation_requests\s*=\s*\[\s*/, content)
+        assert_match(/jwt\.expiration_time\s*=\s*/, content)
+      end
+      assert_migration "db/migrate/create_jwt_denylist.rb"
+      assert_file "app/models/user.rb" do |content|
+        assert_match(/devise :jwt_authenticatable, jwt_revocation_strategy: JwtDenylist/, content)
+      end
+
+      assert_file "app/models/jwt_denylist.rb" do |content|
+        assert_match(/include Devise::JWT::RevocationStrategies::Denylist/, content)
+        assert_match(/self\.table_name = 'jwt_denylist'/, content)
+      end
+    end
+  end
```

Comment:
This test is fine for testing default strategy but we need to also test remaining two strategies in new tests `JTIMatcher, Allowlist`:

test_should_configure_jti_matcher_revocation_strategy
test_should_configure_allowlist_revocation_strategy